pkg/kube/controllers/extendedjob/errand_reconciler.go

```diff
@@ -166,3 +181,62 @@ func (r *ErrandReconciler) createJob(ctx context.Context, extJob ejv1.ExtendedJo
 
 	return nil
 }
+
+// updateSecretReference look and see if any referenced secret is versioned, and if it is, but it's not latest, you have to change the reference
+func (r *ErrandReconciler) updateSecretReference(ctx context.Context, extJob *ejv1.ExtendedJob) error {
```

Comment:
This pattern is defined in multiple places.
Can we move it to the versioned secret store?

Reply:
Yes. It should be.